.circleci/config.yml

```diff
@@ -1,700 +1,655 @@
+            fi
+
+  metrics:
+    <<: *base-job
+    steps:
+      - run:
+          name: Trigger metrics
+          command: |
+            if [ -n "${MOBILE_METRICS_TOKEN}" ]; then
+              if [[ $CIRCLE_BRANCH == main ]]; then
+                curl -X POST --header "Content-Type: application/json" --data "{\"parameters\":{\"run_ios_maps_v10_benchmark\":true,\"ci_ref\":${CIRCLE_BUILD_NUM}},\"branch\":\"master\"}" https://circleci.com/api/v2/project/github/mapbox/mobile-metrics/pipeline?circle-token=${MOBILE_METRICS_TOKEN}
+              else
+                echo "TODO: Triggering metrics from other branches"
+              fi
+            else
+              echo "MOBILE_METRICS_TOKEN not provided"
+            fi
+      - report-failure:
+          report_failure: << parameters.report_failure >>
+          message: "metrics"
+
+  # Run the tests in the MapboxTests scheme
+  maps-unit-tests:
+    <<: *base-job
+    steps:
+      - checkout
+      - access-token-configs
+      - install-dependencies
+      - test-scheme:
+          scheme: MapboxMapsAnnotations
+      - test-scheme:
+          scheme: MapboxMapsFoundation
+      - test-scheme:
+          scheme: MapboxMapsGestures
+      - test-scheme:
+          scheme: MapboxMapsLocation
+      - test-scheme:
+          scheme: MapboxMapsOffline
+      - test-scheme:
+          scheme: MapboxMapsOrnaments
+      - test-scheme:
+          scheme: MapboxMapsSnapshot
+      - test-scheme:
+          scheme: MapboxMapsStyle
+      - test-scheme:
+          scheme: MapboxMaps
+      - report-failure:
+          report_failure: << parameters.report_failure >>
+          message: "maps-unit-tests"
+
+  # We may want to split this across multiple jobs, rather than multiple
+  run-tests-on-devices:
+    <<: *base-job
+    steps:
+      - checkout
+      - access-token-configs
+      - ensure-text-exists-in-commit:
+          commit-text: "[run device tests]"
+          always-run: << parameters.device-tests-always-run >>
+      - run: make install-devicefarm-dependencies
+      - install-dependencies
+      - device-test-scheme:
+          scheme: << parameters.scheme >>
+          app-name: << parameters.app-name >>
+          configuration: Release
+      # get xcresults here, may be zipped with more than one result
+      - gather-xcresults
+      - parse-xcresults
+      - run:
+          name: Symbolicate crash logs
+          command: make symbolicate SCHEME=<< parameters.scheme >> APP_NAME=<< parameters.app-name >> CONFIGURATION=Release
+          when: always
+      - run:
+          name: Converting and uploading coverage
+          command: make device-update-codecov-with-profdata SCHEME=<< parameters.scheme >> APP_NAME=<< parameters.app-name >> CONFIGURATION=Release
+      - store-device-farm-artifacts
+      - store-logs
+      - report-failure:
+          report_failure: << parameters.report_failure >>
+          message: "`<< parameters.scheme >>` device tests"
```

Comment:
If you remove the back-ticks here, we can close https://github.com/mapbox/mapbox-maps-ios/pull/171 😄

Reply:
Updated